COMPONENT_GUIDANCE.md

```diff
@@ -0,0 +1,45 @@
+# Where to add a new component
+
+Problem: You want to add a new component to the Design System.
+The decision you have to make is where to add the component to the project.
+
+- [Where to add a new component](#where-to-add-a-new-component)
+  - [Options](#options)
+  - [Considerations](#considerations)
+    - [Expected Usage](#expected-usage)
+    - [Bundle Size](#bundle-size)
+    - [Design System Upgrade Process](#design-system-upgrade-process)
+  - [Recommendations](#recommendations)
+  - [Naming](#naming)
+
+## Options
+
+There are two places a component could be added to the Design System. The first one is `/packages/core`.
+
+The other option is to create a new package, `/packages/my-new-component`
+
+## Considerations
+
+- expected usage / nature of component
+- bundle size
+- DS upgrade process
+
+### Expected Usage
+
+In general, all of the components in the `core` package are primitives, they are building blocks to be used in constructing more complex UI. Packages that exists outside of `core`, like autocomplete, modal and slider are more complex units of user interface.
+
+### Bundle Size
+
+In the current front-end development enviroment, most projects are able to take advantage of tree-shaking. Tree-shaking is an optimization performed by a bundler (like webpack) to eliminate dead or unused code at build time. It is done by statically analyzing the source code and looking for exports in included packages that are not imported by the application.
```

Comment:
Should we go into more detail about how to enable/utilize tree-shaking?

Reply:
added some more detail